apps/web-staking/src/app/hooks/useGetEsXaiBalance.ts

```diff
@@ -17,7 +18,9 @@ export const useGetEsXaiBalanceHooks = () => {
     });
   }, [address, chainId]);
 
-  return { esXaiBalance };
+  const refreshEsXaiBalance = () => setRefresh((refresh) => !refresh);
+
+  return { esXaiBalance, refreshEsXaiBalance };
```

Comment:
This seems to be missing a listener in the useEffect to trigger something on update the refresh state.

I like having the function returned, but could the function not just reload the esXaiBalance directly ?